### PR TITLE
feat(pre-commit): Add commitizen-branch hook

### PR DESCRIPTION
**Pull request by Kurt-von-Laven**
## Description

Check all commit messages on the current branch. This is useful for checking commit messages after the fact (e.g., pre-push or in CI) since the existing hook only works at commit time. Expand the documentation of the pre-existing commitizen hook to clarify the relationship between them. I wasn't sure whether it would be appropriate to add a section to [this documentation](https://commitizen-tools.github.io/commitizen/auto_check/).

Remove no longer needed `commit-msg` stage from self-use of `commitizen` pre-commit hook in `.pre-commit-config.yaml`. In v2.27.1, the `commitizen` pre-commit hook began specifying that it runs on the `commit-msg` stage in the hook definition in `.pre-commit-hooks.yaml`, so it is no longer necessary to specify the hook stage when using the hook in `.pre-commit-config.yaml`.

## Checklist

- [x] Add test cases to all the changes you introduce
- [x] Run `./scripts/format` and `./scripts/test` locally to ensure this change passes linter check and test
- [x] Test the changes on the local machine manually
- [x] Update the documentation for the changes

## Expected behavior

The `commitizen` hook continues to work as before, allowing empty commit messages. The `commitizen-branch` hook checks all commit messages on your branch, and doesn't allow empty commit messages.

## Steps to Test This Pull Request

1. Replace your Commitizen hook config in your `.pre-commit-config.yaml` with the following:

      ```yaml
      repos:
        - repo: https://github.com/Kurt-von-Laven/commitizen
          rev: pre-commit-hook
          hooks:
            - id: commitizen
            - id: commitizen-branch
              stages: [push]
      ```

1. Ensure that you have pre-commit 1.4.3 or higher installed.
1. Install `commit-msg` and `pre-push` hooks if you haven't already via: `pre-commit install --hook-type commit-msg pre-push`.
1. Commit some changes to your repository.
1. If the commit message was invalid, the `commitizen` hook will fail, and otherwise it will succeed.
1. Commit a change with an empty commit message: `git commit --allow-empty-message -m ""`.
1. Attempt to push your branch.
1. The push fails with an error regarding the empty commit message.

## Additional context
Follows on #512 and #514.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -111,12 +111,14 @@ repos:
...
       - id: commitizen
+      - id: commitizen-branch
+        stages: [push]
 ```
 
 After the configuration is added, you'll need to run
 
 ```sh
-pre-commit install --hook-type commit-msg
+pre-commit install --hook-type commit-msg pre-push
```
Comment (Lee-W) [inline docs/README.md:121]: Should we separate the steps of adding these hooks? e.g., `commitizen` hook might need need `pre-push`

**Review by Lee-W — COMMENTED** [review]
@Woile Overall I'm good with this feature. One of the things come to my mind is whether we really need 2 `cz check` hooks? It seems users can config such behavior on their `.pre-commit-config.yaml` as well

File: docs/README.md (hunk 1 above)
Comment (Kurt-von-Laven) [reply docs/README.md:121]: I clarified the relationship between the hooks and the stages by adding a table. Let me know what you think.

**Comment by Kurt-von-Laven** [comment]
Users can certainly write their own pre-commit hooks, but I consider them tricky to write and test since many implementations I see that haven't already been reviewed by the author of pre-commit get details wrong. I also believe many users will not be aware that the commit-msg hook relies on the developer to have installed the pre-commit hooks correctly since it can't be run in CI. It is a good hook in that it offers real-time feedback, but it is also trivially circumvented (most likely by accident). Tools like [pre-commit.ci](https://pre-commit.ci/), [pre-commit/pre-commit-action](https://github.com/pre-commit/pre-commit-action), and [ScribeMD/pre-commit-action](https://github.com/ScribeMD/pre-commit-action) make it trivial to run the proposed pre-push hook in CI, which is not possible with the existing commit-msg hook.

**Comment by Kurt-von-Laven** [comment]
The test failures seem unrelated to this pull request.

File: docs/README.md (hunk 1 above)
Comment (Lee-W) [reply docs/README.md:121]: It's great! Thanks @Kurt-von-Laven

**Review by Lee-W — APPROVED** [review]
@Woile I'm good with this one. Could you please take a look? Thanks! As for the failed tests, I'll send a separate PR to fix it.

**Pull request merged**